bowparquet.go

```diff
@@ -259,10 +257,84 @@ func (b *bow) WriteParquet(path string, verbose bool) error {
 	return nil
 }
 
-func validateColTypesMeta(b Bow, values string) (colTypesMeta []parquetColTypesMeta, err error) {
-	err = json.Unmarshal([]byte(values), &colTypesMeta)
+// GetParquetMetaColTimeUnit attempts to get the time unit of the column as a time.Duration
+// from the bow metadata read from a parquet file
+// If no time unit metadata is found, time.Duration(0) is returned without error.
+func (b *bow) GetParquetMetaColTimeUnit(colIndex int) (time.Duration, error) {
+	colName := b.ColumnName(colIndex)
+
+	keyIndex := b.Metadata().FindKey(keyParquetColTypesMeta)
+	if keyIndex == -1 {
+		return time.Duration(0), nil
+	}
+
+	colTypesMeta, err := readColTypesMeta(b, b.Metadata().Values()[keyIndex])
+	if err != nil {
+		return time.Duration(0), fmt.Errorf("readColTypesMeta: %w", err)
+	}
+
+	for _, m := range colTypesMeta {
+		if m.Name == colName {
+			if m.LogicalType != nil &&
+				m.LogicalType.IsSetTIMESTAMP() {
+				unit := m.LogicalType.TIMESTAMP.GetUnit()
+				switch {
+				case unit.IsSetMILLIS():
+					return time.Millisecond, nil
+				case unit.IsSetMICROS():
+					return time.Microsecond, nil
+				case unit.IsSetNANOS():
+					return time.Nanosecond, nil
+				}
+			}
+		}
+	}
+
+	return time.Duration(0), nil
+}
+
+// NewMetaWithParquetTimestampCol returns a new bow Metadata from key-value pairs, plus a time column name and its unit as time.Duration.
+// supported durations include time.Millisecond, time.Microsecond and time.Nanosecond.
+func NewMetaWithParquetTimestampCol(keys, values []string, colName string, timeUnit time.Duration) Metadata {
```

Comment:
I actually unexposed and moved this method in `parquet_test.go`, because i don't think it is useful enough.
It was only used there. What do you think?

Reply:
ok for unexposing if not usefull, you can solve